inst/builtin/bs5/shiny/_rules.scss

```diff
@@ -124,3 +124,64 @@ $icon-classes: ("bi", "fa", "fas", "far", "fab", "material-icons") !default;
     --bslib-color-fg: #008558; // Contrast 4.67
   }
 }
+
+// Modals
+.modal-body, .modal-footer {
+  padding: 1.5rem;
+}
+.modal-header .btn-close {
+  align-self: start;
+  display: flex;
+}
+#shiny-modal-wrapper:has( ~ .modal-backdrop) .modal {
+  // Blur the background when the modal has a backdrop
+  backdrop-filter: blur(2px);
+}
+
+// Progress Bars & Notifications
+#shiny-notification-panel {
+  position: fixed;
+  bottom: calc(var(--bslib-spacer, 1rem) / 2);
+  right: calc(var(--bslib-spacer, 1rem) / 2);
+  width: 450px;
+  z-index: $zindex-toast;
+}
+
+.progress-message {
+  margin-right: .6rem;
+}
+
+.shiny-notification {
+  position: relative;
+  opacity: 0.96;
+  padding: calc(var(--bslib-spacer, 1rem) * 2);
+  margin: var(--bslib-spacer, 1rem);
+  border: 1px solid $gray-200;
+  border-radius: 8px;
+  box-shadow: 0px 1px 2px rgba(29, 31, 33, .1), 0px 3px 7px rgba(29, 31, 33, .1), 0px 12px 30px rgba(29, 31, 33, .08);
```

Comment:
Is this dark-mode friendly?

Reply:
I think so, in the shiny preset we use dark shadows that don't change (much) in dark mode. I'd be happy to revisit this again after https://github.com/rstudio/bslib/issues/826

Reply:
I forgot we had added `$box-shadow` and `$box-shadow-dark` into the shiny preset sass. This is done now.